webapp/src/pages/Leaderboard/Leaderboard.js

```diff
@@ -1,9 +1,124 @@
+import React, { useState, useEffect } from "react";
+import { styled } from '@mui/material/styles';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell, { tableCellClasses } from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TableHead from '@mui/material/TableHead';
+import TableRow from '@mui/material/TableRow';
+import Paper from '@mui/material/Paper';
+
+import { ReactComponent as BronzeMedalIcon } from "../../assets/medal-bronze.svg";
+import { ReactComponent as SilverMedalIcon } from "../../assets/medal-silver.svg";
+import { ReactComponent as GoldMedalIcon } from "../../assets/medal-gold.svg";
+
+import { useTranslation } from "react-i18next";
+
+
+const StyledTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+    backgroundColor: theme.palette.common.black,
+    color: theme.palette.common.white,
+  },
+  [`&.${tableCellClasses.body}`]: {
+    fontSize: 14,
+  },
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  '&:nth-of-type(odd)': {
+    backgroundColor: theme.palette.action.hover,
+  },
+  // hide last border
+  '&:last-child td, &:last-child th': {
+    border: 0,
+  },
+}));
+
+function createData(ranking, username, passedQuestions, failedQuestions, gamesPlayed, timePlayed, points) {
+  return { ranking, username, passedQuestions, failedQuestions, gamesPlayed, timePlayed, points };
+}
+
 const Leaderboard = () => {
+
+  //Translation
+  const { t } = useTranslation();
+
+  var [leaderboard, setLeaderboard] = useState([])
+  var [error, setError] = useState(null)
+
+  useEffect(() => {
+    ; (async () => {
+      try {
+        setLeaderboard(await fetchLeaderboard()) // Fetch leaderboard data and set state
+      } catch (error) {
+        setError(error.message) // Set error state if fetch fails
+      }
+    })()
+  }, [])
+
+
+
+  const fetchLeaderboard = async () => {
+    try {
+      const response = await fetch('http://localhost:8000/history/leaderboard');
```

Comment:
Hardcoded url, not suitable for deployment

Reply:
Thank you very much for letting me know, I had the error repeated in several places. Now it's solved.